devito/operator/operator.py

```diff
@@ -585,25 +585,34 @@ def _prepare_arguments(self, autotune=None, **kwargs):
         discretizations = {getattr(kwargs[p.name], 'grid', None) for p in overrides}
         discretizations.update({getattr(p, 'grid', None) for p in defaults})
         discretizations.discard(None)
-        # Remove subgrids if multiple grids
+
+        # Remove subgrids if multiple Grids
```

Comment:
this chunk of code that pulls the grids and sanitizes them to ensure there's just one Grid could/should be now moved to a separate utility function, as I feel it could be reused in some other places as well (surely I found myself in need to extract the unique grid from a list of something... I think IIRC `cluster.grid` is one such case)